### PR TITLE
Added more robust inter-space drone settings.

### PR DESCRIPTION
**Pull request by seankovacs**
**Changes**

- Adds ability to specify row & column inter-spaced drone settings instead of the fixed template layout
- Adds ability to specify the spacing for row & column of inter-spaced drones

### CODE REVIEW

File: src/modules/sbstudio/plugin/operators/create_takeoff_grid.py [hunk 1]
```diff
@@ -224,15 +184,27 @@ class CreateTakeoffGridOperator(Operator):
         update=_ensure_rows_columns_and_counts_consistent,
     )
 
-    drones_per_slot = IntProperty(
-        name="Drones per slot",
+    drones_per_slot_row = IntProperty(
+        name="Drones per slot row",
         description=(
             "Number of drones that can be placed in a single slot in the "
```
Comment (ntamas) [inline src/modules/sbstudio/plugin/operators/create_takeoff_grid.py:190]: Description needs to be updated to reflect that this is now for a single row only.

File: src/modules/sbstudio/plugin/operators/create_takeoff_grid.py [hunk 2]
```diff
@@ -224,15 +184,27 @@ class CreateTakeoffGridOperator(Operator):
...
         soft_min=1,
-        soft_max=len(TAKEOFF_SLOT_TEMPLATES) - 1,
+        soft_max=10,
+        update=_handle_drones_per_slot_change,
+    )
+
+    drones_per_slot_col = IntProperty(
+        name="Drones per slot column",
+        description=(
+            "Number of drones that can be placed in a single slot in the "
```
Comment (ntamas) [inline src/modules/sbstudio/plugin/operators/create_takeoff_grid.py:202]: Description needs to be updated to reflect that this is now for a single column only.

File: src/modules/sbstudio/plugin/operators/create_takeoff_grid.py [hunk 3]
```diff
@@ -262,8 +234,17 @@ class CreateTakeoffGridOperator(Operator):
         unit="LENGTH",
     )
 
-    intra_slot_spacing = FloatProperty(
-        name="Intra-slot spacing",
+    intra_slot_spacing_row = FloatProperty(
+        name="Intra-slot spacing row",
+        description="Spacing between drones in the same slot of the grid",
```
Comment (ntamas) [inline src/modules/sbstudio/plugin/operators/create_takeoff_grid.py:239]: Description needs to be updated to reflect that this is now for a single row only.

File: src/modules/sbstudio/plugin/operators/create_takeoff_grid.py [hunk 4]
```diff
@@ -262,8 +234,17 @@ class CreateTakeoffGridOperator(Operator):
...
+        default=0.5,
+        soft_min=0,
+        soft_max=5,
+        unit="LENGTH",
+    )
+
+    intra_slot_spacing_col = FloatProperty(
+        name="Intra-slot spacing column",
         description="Spacing between drones in the same slot of the grid",
```
Comment (ntamas) [inline src/modules/sbstudio/plugin/operators/create_takeoff_grid.py:248]: Description needs to be updated to reflect that this is now for a single column only.

**Review by ntamas — CHANGES_REQUESTED** [review]
Thanks for the contribution, it looks good to me, but the descriptions of the newly introduced properties need to be updated (see per-line comments).

**Comment by seankovacs** [comment]
Updated the naming and descriptions. Let me know if it makes sense

**Review by ntamas — APPROVED** [review]

**Pull request merged**